charger/openevse.go

```diff
@@ -134,72 +111,39 @@ func (c *OpenEVSE) IsChargingOnThreePhases() (threePhase bool, err error) {
 	return threePhaseInt != 0, err
 }
 
-// func (c *OpenEVSE) Phases() (phases int, err error) {
-// 	if c.phases != 0 {
-// 		return c.phases, nil
-// 	}
-
-// 	isChargingOnThreePhase, err := c.IsChargingOnThreePhases()
-// 	if isChargingOnThreePhase {
-// 		phases = 3
-// 	} else {
-// 		phases = 1
-// 	}
-
-// 	return phases, err
-// }
-
 func (c *OpenEVSE) PerformRAPICommand(uri, command string) (response string, success bool, err error) {
 	var uriBuilder strings.Builder
 	uriBuilder.WriteString(uri)
 	uriBuilder.WriteString("/r?json=1&rapi=")
 	uriBuilder.WriteString(url.QueryEscape(command))
 
-	resp, err := c.helper.Get(uri)
+	var resp openevse.RAPIResponse
+	err = c.helper.GetJSON(uriBuilder.String(), &resp)
 	if err != nil {
+		fmt.Printf("PerformRAPICommand %s: %+v", command, resp)
 		return "", false, err
 	}
 
-	bodyBytes, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return "", false, err
+	regex := regexp.MustCompile(`\$(OK|NK)([^^]*)\^\d+`)
+	if regex == nil {
+		return "", false, fmt.Errorf("internal error")
 	}
-	defer resp.Body.Close()
-
-	switch {
-	case strings.Contains(resp.Header.Get("Content-Type"), "json") && resp.StatusCode == 200:
-		var responseJson struct {
-			Cmd *string
-			Ret *string
-		}
-		if err := json.Unmarshal(bodyBytes, &responseJson); err != nil {
-			return "", false, err
-		}
 
-		regex := regexp.MustCompile(`\$(OK|NK)([^^]*)\^\d+`)
+	matches := regex.FindStringSubmatch(resp.Ret)
```

Comment:
Didn't you say it's json? No need for regexp then!

Reply:
This is the actual response: 
```JSON
{
  "cmd": "$G7",
  "ret": "$OK 1^31"
}
```
`ret` can be either `$OK` or `$NK`, followed by a value then a CRC.

Reply:
Then please:

```
var res struct{
	Cmd, Ret string
}

err := c.GetJSON(uri, &res)

if err == nil && strings.HasPrefix(res.Ret, "$OK") {
    ...
}

```

Reply:
I still need to parse the value. Do you have a better way?

Reply:
Just split it at the space?